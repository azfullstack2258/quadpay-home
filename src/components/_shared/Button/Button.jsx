import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { noop } from 'utils/helper';
import Icons from '../Icons';

const Btn = styled.button`
  svg {
    margin-right: 10px;
  }
`;

const Button = ({ className: cls, caption, type, disabled, icon, onClick }) => {
  const className = useMemo(
    () => `rounded px-6 py-3 flex items-center bg-white text-blue font-bold ${disabled ? 'btn-disabled' : `btn-${type}`} ${cls}`,
    [type, disabled]
  );
  const Icon = icon !== '' ? Icons[icon] : null;

  return (
    <Btn className={className} onClick={onClick}>
      {Icon && <Icon />}
      {caption}
    </Btn>
  );
};

Button.propTypes = {
  caption: PropTypes.string,
  type: PropTypes.string, // primary, secondary, neutral, flat
  disabled: PropTypes.bool,
  icon: PropTypes.string,
  onClick: PropTypes.func,
};

Button.defaultProps = {
  caption: '',
  type: 'primary',
  disabled: false,
  icon: '',
  onClick: noop,
};

export default Button;
