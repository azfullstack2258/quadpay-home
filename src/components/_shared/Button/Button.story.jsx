import React from 'react';
import { storiesOf } from '@storybook/react';

import Example from '../../../utils/storybook-wrapper';
import Stack from '../../../utils/storybook';

import Button from './Button';

storiesOf('Components/Button', module)
  .add('appearance', () => (
    <Example title="Button appearance">
      <Stack>
        <Button type="primary" caption="Primary button" />
        <Button type="secondary" caption="Secondary button" />
        <Button type="neutral" caption="Neutral button" />
        <Button type="flat" caption="Flat button" />
        <Button type="primary" caption="Disabled button" disabled />
      </Stack>
    </Example>
  ))
  .add('with icon', () => (
    <Example title="Button appearance">
      <Stack>
        <Button type="primary" caption="Primary button" icon="temple" />
        <Button type="secondary" caption="Secondary button" icon="temple" />
        <Button type="neutral" caption="Neutral button" icon="temple" />
        <Button type="flat" caption="Flat button" icon="temple" />
        <Button
          type="primary"
          caption="Disabled button"
          icon="temple"
          disabled
        />
      </Stack>
    </Example>
  ));
