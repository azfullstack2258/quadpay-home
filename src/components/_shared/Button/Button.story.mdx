<!--- Button.story.mdx -->

import { Meta, Story, Canvas } from '@storybook/addon-docs/blocks';
import { Button } from './Button';

<Meta title="MDX/Button" component={Button} />

# Button
Button component

export const Template = (args) => <Button {...args} />

<Canvas>
  <Story name="Primary" args={{ type: 'primary', caption: 'Primary' }}>
    {Template.bind({})}
  </Story>
  <Story name="Secondary" args={{ type: 'secondary' }}>
    {Template.bind({})}
  </Story>
  <Story name="Neutral" args={{ type: 'neutral', caption: 'Neutral' }}>
    {Template.bind({})}
  </Story>
  <Story name="Flat" args={{ type: 'flat', caption: 'Flat' }}>
    {Template.bind({})}
  </Story>
  <Story name="Disabled" args={{ disabled: true, caption: 'Disabled' }}>
    {Template.bind({})}
  </Story>
</Canvas>
