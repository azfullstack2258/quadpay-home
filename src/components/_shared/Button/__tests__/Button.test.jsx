import React from 'react';
import { render } from '@testing-library/react';
import Button from '../Button';

describe('<Button />', () => {
  it('should render a prop', () => {
    const { container, getByText } = render(
      <Button type="primary" caption="Primary button" />
    );

    expect(container).toMatchSnapshot();
    expect(container.firstChild.classList.contains('btn-primary')).toBe(true);

    const buttonElement = getByText(/primary button/i);
    expect(buttonElement).toBeInTheDocument();
  });
});
