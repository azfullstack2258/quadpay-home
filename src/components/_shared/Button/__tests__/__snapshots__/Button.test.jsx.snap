// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render a prop 1`] = `
<div>
  <button
    class="btn btn-primary"
  >
    Primary button
  </button>
</div>
`;
